Document metrics and update overview section (#1950)
prometheus/docs: content/docs/introduction/overview.md

* targets are discovered via service discovery or static configuration
* multiple modes of graphing and dashboarding support

### What are metrics ?

> In layperson terms, metrics are a standard for measurement. What we want to measure differs from application to application. For a web server it can be request times, for a database it can be number of active connections or number of active queries etc.

Reviewer: ```suggestion
In layperson terms, metrics are numeric measurements, time series mean that changes are recorded over time. What users want to measure differs from application to application. For a web server it might be request times, for a database it might be number of active connections or number of active queries etc.
```

> Metrics play an important role in understanding why your application is working in a certain way. Lets assume you are running a web application and find that the application is slow. You will need some information to find out what is happening with your application. For example the application can become slow when the number of requests are high. If you have the request count metric you can spot the reason and increase the number of servers to handle the heavy load. 

Reviewer: ```suggestion
Metrics play an important role in understanding why your application is working in a certain way. Lets assume you are running a web application and find that the application is slow. You will need some information to find out what is happening with your application. For example the application can become slow when the number of requests are high. If you have the request count metric you can spot the reason and increase the number of servers to handle the load. 
```

> Whenever you are defining the metrics for your application you must put on your detective hat and ask this question what all information will be important for me to debug if any issue occurs in my application?

Reviewer: That feels weird in an intro section.

> 
> Whenever you are defining the metrics for your application you must put on your detective hat and ask this question what all information will be important for me to debug if any issue occurs in my application?
> 

Reviewer: ```suggestion
```

* targets are discovered via service discovery or static configuration
* multiple modes of graphing and dashboarding support

### What are metrics ?

> Metrics in layman terms is a standard for measurement. What we want to measure differs from application to application. For a web server it can be request times, for a database it can be number of active connections or number of active queries etc.

Reviewer 2: s/layman/layperson

> Metrics play an important role in understanding why your application is working in a certain way. Lets assume you are running a web application and find that the application is slow. You will need some information to find out what is happening with your application. For example the application can become slow when the number of requests are high. If you have the request count metric you can spot the reason and increase the number of servers to handle the heavy load. Whenever you are defining the metrics for your application you must put on your detective hat and ask this question what all information will be important for me to debug if any issue occurs in my application?

Reviewer 2: I wonder if it would help here if we would be more explicit with the example. So, rather than talking in general about a web app and requests, how about a concrete example using HTTP API, GET requests, etc.?

> Metrics play an important role in understanding why your application is working in a certain way. Lets assume you are running a web application and find that the application is slow. You will need some information to find out what is happening with your application. For example the application can become slow when the number of requests are high. If you have the request count metric you can spot the reason and increase the number of servers to handle the heavy load. Whenever you are defining the metrics for your application you must put on your detective hat and ask this question what all information will be important for me to debug if any issue occurs in my application?

Author: @mhausenblas I think giving a simple yet generic example will be more apt than getting into something specific.
